tabular/src/autogluon/tabular/models/ebm/ebm_model.py

```diff
@@ -0,0 +1,241 @@
+from __future__ import annotations
+
+import time
+import warnings
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pandas as pd
+from autogluon.core.constants import BINARY, MULTICLASS, REGRESSION
+from autogluon.core.models import AbstractModel
+
+from .hyperparameters.parameters import get_param_baseline
+from .hyperparameters.searchspaces import get_default_searchspace
+
+if TYPE_CHECKING:
+    from autogluon.core.metrics import Scorer
+
+
+class EbmCallback:
+    """Time limit callback for EBM."""
+
+    def __init__(self, seconds: float):
+        self.seconds = seconds
+        self.end_time: float | None = None
+
+    def __call__(self, *args, **kwargs):
+        if self.end_time is None:
+            self.end_time = time.monotonic() + self.seconds
+            return False
+        return time.monotonic() > self.end_time
+
+
+class EBMModel(AbstractModel):
```

Comment:
Might be good to include a docstring describing the model (refer to the docstrings I added for the other recent models such as TabPFNv2, TabM, RealMLP, etc.)

Reply:
done